roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -140,40 +154,26 @@ def is_available_skopeo_image(self, image, registry, task_vars):
             image=image,
         )
 
-        args = {
-            "name": "skopeo_inspect",
-            "image": self.skopeo_image,
-            "command": cmd_str,
-            "detach": False,
-            "cleanup": True,
-        }
-        result = self.module_executor("docker_container", args, task_vars)
-        return result.get("failed", False)
-
-    def containerized_docker_images(self, base_name, version):
-        return [
-            "{image}:{version}".format(image=self.image_from_base_name(base_name), version=version)
-        ]
+        args = {"_raw_params": cmd_str}
+        result = self.module_executor("command", args, task_vars)
+        return not result.get("failed", False) and result.get("rc", 0) == 0
 
     @staticmethod
-    def rpm_docker_images(base, version):
-        return [
-            "{image_repo}/registry-console:{version}".format(image_repo=base["repo"], version=version)
-        ]
+    def non_qualified_docker_images(namespace, name, version, is_containerized):
+        if is_containerized:
+            return ["{}/{}:{}".format(namespace, name, version)] if name else []
+
+        return ["{}/{}:{}".format(namespace, name, version)] if name else []
 
     @staticmethod
-    def qualified_docker_images(image_name, version):
+    def qualified_docker_images(namespace, name, version):
         return [
-            "{}-{}:{}".format(image_name, component, version)
-            for component in "haproxy-router docker-registry deployer pod".split()
+            "{}/{}-{}:{}".format(namespace, name, suffix, version)
+            for suffix in ["haproxy-router", "docker-registry", "deployer", "pod"]
         ]
 
-    @staticmethod
-    def image_from_base_name(base):
-        return "".join([base["repo"], "/", base["image"]])
-
     # ensures that the skopeo docker image exists, and updates it
     # with latest if image was already present locally.
     def update_skopeo_image(self, task_vars):
-        result = self.module_executor("docker_image", {"name": self.skopeo_image}, task_vars)
-        return result.get("msg", ""), result.get("failed", False), result.get("changed", False)
+        result = self.module_executor("yum", {"name": "skopeo", "state": "latest"}, task_vars)
```

Comment:
@sosiouxme went ahead and removed `docker-py` dep for using skopeo, however [this line](https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_health_checker/openshift_checks/docker_image_availability.py#L50) will execute before the skopeo check, to retrieve known docker registries. Which means we will still fail without `docker-py` before even hitting the skopeo part of this check. 

A possible solution would be to have a default `docker.io` registry, so that in case no registries are found, we just use that one (a warning here explaining what happened would most likely be in order, however it might be something that ultimately gets lost in a wall of text.

Another solution could be to try to look in `ansible_facts` for `additional_registries` under `ansible_facts.openshift.docker`